rust/api.md

```diff
@@ -0,0 +1,532 @@
+# Definition for the new Scanner HTTP API via JSON
+
+
+## Requests
+
+### start scan
+
+The start scan command is used to start a scan. It is important to use a unique scan ID. The response of this request just returns a status code wether it worked or not, without any data.
+
+- Path: `/scan/start`
+- Request:
+<pre>
+  <code>
+    {
+      <a href="#scan_id">"scan_id"</a>: string,
+      <a href="#targets">"targets"</a>: [{
+        <a href="#target">"target"</a>: [string],
+        <a href="#ports">"ports"</a>: [string],
+        <a href="#credentials">"credentials"</a>: [{
+          <a href="#service">"service"</a>: string,
+          <a href="#port">"port"</a>: int,
+          <a href="#up">"up"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string
+          },
+          <a href="#usk">"usk"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string,
+            <a href="#private">"private"</a>: string
+          },
+          <a href="#snmp">"snmp"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string,
+            <a href="#community">"community"</a>: string,
+            <a href="#auth_algorithm">"auth_algorithm"</a>: string,
+            <a href="#privacy_password">"privacy_password"</a>: string,
+            <a href="#privacy_algorithm">"privacy_algorithm"</a>: string
+          }
+        }]
+      }],
+      <a href="#excluded">"excluded"</a>: [string],
+      <a href="#scanner_parameters">"scanner_parameters"</a>: {key:value},
+      <a href="#vts">"vts"</a>: {
+        <a href="#vt_single">"vt_single"</a>: [{
+          <a href="#oid">"oid"</a>: string,
+          <a href="#vt_parameters">"vt_parameters"</a>: [{
+            <a href="#parameter">"parameter"</a>: string,
+            <a href="#value">"value"</a>: any
+          }]
+        }],
+        <a href="#vt_group">"vt_group"</a>: [string]
+      }
+    }
+  </code>
+</pre>
+- Status codes: 
+  - 200 Ok
+  - 400 Bad Request
+- Response: None
+
+
+### get scan
+
+The get scan command returns information about a given scan. By default only basic information about the scan are requested. With the fields details the command also returns results.
+
+- Path: `/scan/get`
```

Comment:
I would never use a action in a path since this is already expressed by the http call GET and than I wouldn't require the client to use GET with sending a json but use path parameter instead.

e.g.
```
GET /scan/{id}?pop_resuls=true&host_info=...
```

Reply:
Not sure how we should handle the GET command with the pop_results. Normally the GET method should only be used to retrieve data, but I guess it is more a guideline than an obligation. But nonetheless it seems a bit dirty.

Reply:
true, maybe the whole pop is a thing of a past and we should require a proper delete call?

Reply:
That could be rather dangerous, if exactly between a get scan and delete results call, new results are generated, we would miss them

Reply:
if the client tries to delete why the scan in running and not stopped we should just not allow it?

Reply:
Currently the pop results is just there due to memory limitation if we store on io or db either way I don't think deleting results while the scan is still running is as necessary as in the current setup.

Reply:
Considering that we use PATCH for stopping we could also use PATCH to pop results and return the popped results.

With that we would have standard GET and if somebody for some reasons think he wants to pop results while a scan is running he could use PATCH with something like

```
{
"pop": true
}
```

on `/scan/{id}`.

Reply:
Yes that sounds good

Reply:
I am just thinking about moving the management of scans to the scanner, as we wanted to do it with eulabeia. That would allow to differentiate things a bit more and does not combine distinct commands like create and start scan. What do you think about it?

Reply:
As long as we keep in mind that it has to be easy to integrate in existing systems, that may already have plugin selection, scheduling etc, I think it is a good idea.